src/averageAges.js

```diff
@@ -37,9 +45,23 @@
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womens = people.filter(person => person.sex === 'f');
+
+  if (withChildren === true) {
+    womens = womens.filter(woman => people.some(person =>
+      person.mother === woman.name));
+  }
+
+  const totalAge = womens.reduce((sum, person) => sum
+    + (person.died - person.born), 0);
+
+  const averageAge = parseFloat((totalAge / womens.length).toFixed(2));
+
+  return averageAge;
```

Comment:
Your implementation correctly calculates the average age for women, with or without children. However, consider using `=== true` in the condition (line 50) and `=== false` for better readability. For example, `if (withChildren === true)` can be simplified to `if (withChildren)`, and `if (!withChildren)` can be used for the opposite condition.